### PR TITLE
[Tech] Refactor game settings with fallbacks 

### PR DESCRIPTION
**Pull request by arielj**
This PR should fix the 2 issues mentioned here https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/1491 by adding more customization around which settings are stored for a game depending on the platform. I think there's still room for improvement there (like... gamemode, mongohud, audio fix, and more can be moved to only linux), but I wanted to fix the issue related to the wine config first before moving more things around and have more tests.

The issues I noticed in the original code:
- we were defaulting to the complete global settings if no config file for a game, this is not necesarilly bad but we just dumped everything there without any sanity check
- we were trying to do a `winePrefix.replace` even on windows on mac, which is not only not needed but also failed in some cases when `winePrefix` was not a string
- we were storing extra information for all platforms, which makes reading logs kinda confusing sometimes

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: electron/game_config.ts [hunk 1]
```diff
@@ -252,19 +246,36 @@ class GameConfigV0 extends GameConfig {
...
-    return settingsToReturn
+    // set specific keys depending on the platform
+    if (isMac) {
+      defaultSettings.wineCrossoverBottle = wineCrossoverBottle
+    } else if (isLinux) {
+      defaultSettings.wineVersion = wineVersion
+      defaultSettings.winePrefix = winePrefix || `${userHome}/.wine`
+
+      // fix winePrefix if needed
+      if (gameSettings.winePrefix) {
```
Comment (flavioislima) [inline electron/game_config.ts:269]:
```suggestion
      if (gameSettings.winePrefix?.includes('~')) {
```
so maybe we can avoid doing replace always?

**Review by flavioislima — APPROVED** [review]
Looks good, just one small comment.

**Comment by arielj** [comment]
I was testing it more on a Mac and I broke something, I won't merge this for now, I can't ignore `wineVersion` on mac like I was doing here

**Review by flavioislima — APPROVED** [review]
LGTM

**Pull request merged**